### ISSUE TITLE
Official Gov Guidelines for Progressive Enhancement

### ISSUE THREAD

**Issue by nelsonic** [nelsonic/learn-progressive-enhancement#6]
https://www.gov.uk/service-manual/making-software/progressive-enhancement

**Comment by iteles** [comment]
https://gdstechnology.blog.gov.uk/2016/09/19/why-we-use-progressive-enhancement-to-build-gov-uk/ 

Thanks for sharing @benjaminlees 👍 :arrow_up:

**Comment by nelsonic** [comment]
Yeah, the GOV.UK progressive enhancement article was at the _Top_ of Hacker News yesterday: 
![progressive-enhancement-top-of-hacker-news-sept-2016](https://cloud.githubusercontent.com/assets/194400/18703050/3762f07c-7fdc-11e6-9643-189360c47686.png)
Accompanied by the usual comment-war https://news.ycombinator.com/item?id=12538144
In which _lazy/ignorant_ people attempt to make the case why there is "_no time_" to build things _progressively_ ...

Another good link is to the GOV.UK Design Principals: https://www.gov.uk/design-principles which I _highly_ doubt most "_Modern JavaScript_" developers have _read_ let alone _follow_...

Because as we _all_ know, "_Progressive Enhancement is Dead_": http://tomdale.net/2013/09/progressive-enhancement-is-dead/ 

> A _**better question**_ is: When do we _not_ want to support 100% of users...?

Most people don't make a _**conscious decision**_ to _not_ support "No JS", they do it by _default_ because Front-end Frameworks make the _**implicit assumption**_ that _all_ web apps should be built in the browser and thus _require_ JS ... an entire _generation_ of web developers are entering the "_market_" not _knowing_ the _messy reality_ that _Millions_ of people are still stuck on "_Old_" devices/browsers and often don't have the _option_ of upgrading.

Meanwhile **_Google**_ is still flying the flag for Progressive Enhancement with their AMP & PWA projects, and we're _stoked_ that _someone_ is still taking the time to cater for _everyone_.

**Comment by aarongustafson** [comment]
Wait, you mean this Tom Dale? ;-)

http://tomdale.net/2015/02/youre-missing-the-point-of-server-side-rendered-javascript-apps/

He did a complete 180, but of course never mentioned that Fastboot was PE.

**Comment by nelsonic** [comment]
@aarongustafson indeed. a 180<sup>o</sup> turn from 2013 --> 2015 ... I wonder _why_...? 😕

**Comment by aarongustafson** [comment]
Some jaded colleagues have speculated that he was exacerbating a problem in advance of "selling" his own solution. I like to think he realized the error of his ways. I suspect it's the latter simply because he never full owns up and says "Hey you know that thing I said about progressive enhancement being nonsense? I was totally wrong. If you, like me, have gone down the route of SPAs with reckless abandon, I have a solution that will help get you progressive enhancement…”
